lib/publishing_api_finder_publisher.rb

```diff
@@ -98,12 +99,11 @@ def initialize(schema, timestamp)
 
   def details
     details = schema.fetch("details", {})
-    details.merge(
+    details.slice(GovukSchemas::Schema.find(publisher_schema: "finder_email_signup"))
```

Comment:
It might be nice to put this part in a separate method, so that you can explain what it does in a method name. Does `details.slice(schema)` give you the details hash with unpermitted keys removed?

Reply:
Added a comment to better explain what's happening here.